src/index.js

```diff
@@ -176,37 +182,49 @@ function addMetadata({ types: t }) {
       CallExpression(babelPath, state) {
         if (!state.opts.shouldLoadFile) return;
 
-        // If we're at a top-level call expression, then we reset the beforeEachModified state to false, and let Babel
-        // move on to the next call expression.
-        if (babelPath.scope.block.type === 'Program') {
-          state.opts.beforeEachModified = false;
-          return;
+        // If this call expression is a top-level module, store string arg, else skip the nested module entirely
+        if (babelPath.get('callee').isIdentifier({ name: 'module' })) {
+          if (babelPath.parentPath.parent.type === 'Program') {
+            state.opts.moduleName = babelPath.get('arguments')[0].node.value;
+          } else {
+            // skip traversing contents in this nested module
+            babelPath.skip();
+          }
         }
 
-        if (!state.opts.beforeEachModified) {
-          const hasBeforeEach = babelPath.node.callee.property
+        if (
+          state.opts.moduleName &&
+          !state.opts.transformedModules.includes(state.opts.moduleName)
+        ) {
+          const isBeforeEach = babelPath.node.callee.property
             ? babelPath.node.callee.property.name === 'beforeEach'
             : false;
-          const testMethodCalls = ['test', 'module'];
-
-          let nodeName = '';
-
-          if (babelPath.node.callee.name) {
-            nodeName = babelPath.node.callee.name;
-          } else if (babelPath.node.callee.object) {
-            nodeName = babelPath.node.callee.object.name;
+          let isFirstChildTestMethodCall;
+
+          if (!isBeforeEach) {
+            const testMethodCalls = ['test', 'tests', 'module'];
+            let nodeName = '';
+
+            if (babelPath.node.callee && babelPath.node.callee.name) {
+              nodeName = babelPath.node.callee.name;
+            } else if (babelPath.node.callee.object) {
+              nodeName = babelPath.node.callee.object.name;
+            } else {
+              nodeName = babelPath.node.name;
+            }
+
+            isFirstChildTestMethodCall =
+              nodeName &&
+              testMethodCalls.includes(nodeName) &&
+              babelPath.scope.path.parentPath &&
+              babelPath.scope.path.parentPath.node &&
+              babelPath.scope.path.parentPath.node.callee &&
+              babelPath.scope.path.parentPath.node.callee.name === 'module';
```

Comment:
This drilling into properties feels like a smell to me. I'm not sure yet what approach would make this more succinct and DRY..

Reply:
The only way to do this would to use a helper that allows for inspecting a path. 

Example:

```js
function get(node, path) {
  if (!node) {
    return;
  }

  let parts;
  if (typeof path === 'string') {
    parts = path.split('.');
  } else {
    parts = path;
  }

  if (parts.length === 1) {
    return node[path];
  }

  let property = node[parts[0]];

  if (property && parts.length > 1) {
    parts.shift();
    return get(property, parts);
  }

  return property;
}
```

Which you'd call like so:

```js
get(babelPath, 'scope.path.parentPath.node.callee.name')
```